mir_eval/transcription.py

```diff
@@ -353,7 +523,182 @@ def precision_recall_f1(ref_intervals, ref_pitches, est_intervals, est_pitches,
+    -------
+    avg_overlap_ratio: float
+        The computed Average Overlap Ratio score
+    """
+    ratios = []
+    for match in matching:
+        ref_int = ref_intervals[match[0]]
+        est_int = est_intervals[match[1]]
+        overlap_ratio = (
+            (min(ref_int[1], est_int[1]) - max(ref_int[0], est_int[0])) /
+            (max(ref_int[1], est_int[1]) - min(ref_int[0], est_int[0])))
+        ratios.append(overlap_ratio)
+
+    if len(ratios) == 0:
+        return 0
+    else:
+        return np.mean(ratios)
+
+
+def onset_precision_recall_f1(ref_intervals, est_intervals,
+                              onset_tolerance=0.05, strict=False):
+    """Compute the Precision, Recall and F-measure of note onsets: an estimated
+    onset is considered correct if it is within +-50ms of a ref onset. Note
+    that this metric completely ignores note offset and note pitch. This means
+    an estimated onset will be considered correct if it matches a
+    reference onset, even if the onsets come from notes with completely
+    different pitches (i.e. notes that would not match with `match_notes`).
+
+
+    Examples
+    --------
+    >>> ref_intervals, _ = mir_eval.io.load_valued_intervals(
+    ...     'reference.txt')
+    >>> est_intervals, _ = mir_eval.io.load_valued_intervals(
+    ...     'estimated.txt')
+    >>> (onset_precision,
+    ...  onset_recall,
+    ...  onset_f_measure) = mir_eval.transcription.onset_precision_recall_f1(
+    ...      ref_intervals, est_intervals)
+
+    Parameters
+    ----------
+    ref_intervals : np.ndarray, shape=(n,2)
+        Array of reference notes time intervals (onset and offset times)
+    est_intervals : np.ndarray, shape=(m,2)
+        Array of estimated notes time intervals (onset and offset times)
+    onset_tolerance : float > 0
+        The tolerance for an estimated note's onset deviating from the
+        reference note's onset, in seconds. Default is 0.05 (50 ms).
+    strict: bool
+        If ``strict=False`` (the default), threshold checks for onset matching
+        are performed using ``<=`` (less than or equal). If ``strict=True``,
+        the threshold checks are performed using ``<`` (less than).
+
+    Returns
+    -------
+    precision : float
+        The computed precision score
+    recall : float
+        The computed recall score
+    f_measure : float
+        The computed F-measure score
+    """
+    validate_intervals(ref_intervals, est_intervals)
+    # When reference notes are empty, metrics are undefined, return 0's
+    if len(ref_intervals) == 0 or len(est_intervals) == 0:
+        return 0., 0., 0.
+
+    matching = match_onsets(ref_intervals, est_intervals,
+                            onset_tolerance=onset_tolerance,
+                            strict=strict)
+
+    onset_precision = float(len(matching))/len(est_intervals)
+    onset_recall = float(len(matching))/len(ref_intervals)
+    onset_f_measure = util.f_measure(onset_precision, onset_recall)
```

Comment:
The other parts of mir_eval that do this sort of prf calculation also accept a `beta` parameter to balance the F-score calculation.  I think for consistency, this should be supported in the transcription metrics as well.